### PR TITLE
fix(compiler): Report Win32 instead of Cygwin in JS compiler

### PR DESCRIPTION
**Pull request by phated**
The JS version of the compiler kept crashing because it was unable to locate the stdlib. This was due to JSOO reporting itself as "Cygwin" instead of "Win32". We've pushed for a lot of upstream changes in `Fp` so we use the correct separators on Windows, but that doesn't happen if it thinks it is running in Cygwin.

This hack (placed in `Grain_utils` so pretty much everything will get it applied) will swap us back to reporting `Win32` and make the all our tools work.

### CODE REVIEW

**Review by ospencer — APPROVED** [review]

**Review by marcusroberts — APPROVED** [review]

**Comment by phated** [comment]
Looks like this might be causing a bug in the relative import resolution. I'll need to debug before we can land it.

File: compiler/src/utils/filepath.re [hunk 1]
```diff
@@ -85,6 +87,44 @@ module String = {
...
+
+  // TODO(#216): We should consider switching to type safe Fp.t where ever filepaths are used
+  let dirname = path => {
+    normalize_separators(Filename.dirname(path));
+  };
+
+  // TODO(#216): We should consider switching to type safe Fp.t where ever filepaths are used
+  let basename = Filename.basename;
+
+  // TODO: This is poorly named
```
Comment (ospencer) [inline compiler/src/utils/filepath.re:114]: Do we want to just rename now?

**Review by ospencer — APPROVED** [review]
New version looks great!

File: compiler/src/utils/filepath.re (hunk 1 above)
Comment (phated) [reply compiler/src/utils/filepath.re:114]: Yeah. Any suggestions? It's like `non_module_relative_path`? 🤣

Comment (ospencer) [reply compiler/src/utils/filepath.re:114]: Could go the OCaml approach and name it `is_really_relative` 😛

Comment (phated) [reply compiler/src/utils/filepath.re:114]: God no

Comment (phated) [reply compiler/src/utils/filepath.re:114]: Let's skip for now. We probably want to change the way modules are specified when we completely change the module system.

**Pull request merged**